### DISCUSSION TITLE
It is possible to enable Deno unstable flag on Edge Functions?

### DISCUSSION THREAD

**Discussion by thawankeane** [supabase/supabase#14604] [Questions]
I need to make authenticated fetch API calls through TLS using certificates, the only way that I found to make it is to use the unstable [`Deno.createHttpClient`](https://deno.land/api@v1.34.0?unstable&s=Deno.createHttpClient) function. It works locally, but when I deployed to supabase at runtimes it throws an error `Deno.createHttpClient is not a function `.

My guess is that it may be that supabase edge functions by default do not come with the `unstable` flag as `true`. Is there a way to make this work?

**Comment by mertyildiran** [comment]
@thawankeane Supabase uses [Deno Deploy](https://deno.com/deploy) according to [this blog post](https://supabase.com/blog/edge-runtime-self-hosted-deno-functions).

By following these discussions and issues:
- https://github.com/denoland/deno/discussions/13607
- https://github.com/denoland/deploy_feedback/issues/168
- https://github.com/denoland/deploy_feedback/issues/22

I found out Deno Deploy is a different runtime than Deno and things like [`Deno.createHttpClient`](https://deno.land/api@v1.36.4?s=Deno.createHttpClient&unstable=) from the unstable API, are not supported in Deno Deploy. I wish that was not the case, because I really need [`Deno.createHttpClient`](https://deno.land/api@v1.36.4?s=Deno.createHttpClient&unstable=) in my Edge Functions.

**Reply by jrnkng** [reply]
Did you ever solve this? Deno KV is supported in Deno Deploy but still need to use unstable flag. Not sure how

**Comment by ardabeyazoglu** [comment]
Self-hosted services usually use self-signed certificates with self-signed root certificates. Deno connectTls does not give the option to trust them, which makes us receive the following error in this case. There should be a way of connecting to servers with self-signed certificates i think.

error: Uncaught (in promise) InvalidData: invalid peer certificate: Other(OtherError(UnsupportedCertVersion))

"--unsafely-ignore-certificate-errors" would solve it, but they wouldn't enable this for security. I think there needs to be a way to trust based on specific connection.
